src/java.base/share/classes/java/text/BreakIterator.java

```diff
@@ -560,21 +560,13 @@ private static BreakIterator createBreakInstance(Locale locale,
 
     private static BreakIterator createBreakInstance(LocaleProviderAdapter adapter, Locale locale, int type) {
         BreakIteratorProvider breakIteratorProvider = adapter.getBreakIteratorProvider();
-        BreakIterator iterator = null;
-        switch (type) {
-        case CHARACTER_INDEX:
-            iterator = breakIteratorProvider.getCharacterInstance(locale);
-            break;
-        case WORD_INDEX:
-            iterator = breakIteratorProvider.getWordInstance(locale);
-            break;
-        case LINE_INDEX:
-            iterator = breakIteratorProvider.getLineInstance(locale);
-            break;
-        case SENTENCE_INDEX:
-            iterator = breakIteratorProvider.getSentenceInstance(locale);
-            break;
-        }
+        BreakIterator iterator = switch (type) {
+            case CHARACTER_INDEX -> breakIteratorProvider.getCharacterInstance(locale);
+            case WORD_INDEX      -> breakIteratorProvider.getWordInstance(locale);
+            case LINE_INDEX      -> breakIteratorProvider.getLineInstance(locale);
+            case SENTENCE_INDEX  -> breakIteratorProvider.getSentenceInstance(locale);
+            default              -> null;
+        };
```

Comment:
No need to use the local variable `iterator` here. Simply return with the switch expression.

Reply:
Hi Naoto, thanks for spotting this. Code updated as suggested. See adc8af4